### DISCUSSION TITLE
Installazione di Docker compose

### DISCUSSION THREAD

**Discussion by Robertomastro** [indomus/forum#1186] [HUB: Home Assistant]
buongiorno,
seguendo la guida: 

Per installare anche il tool Docker Compose (consigliato), eseguire i comandi:

python3 -m pip install docker-compose

sono incorso nelle seguenti segnalazione:

  WARNING: The script dotenv is installed in '/home/pi/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script jsonschema is installed in '/home/pi/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
  WARNING: The script docker-compose is installed in '/home/pi/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.

posso non considerarle? grazie

**Comment by snakuzzo** [comment]
Se esci dalla sessione SSH e ti ricolleghi (o se ricarichi il .profile/.bashrc) non dovresti più avere quei warning.

P.S. Per favore, per queste semplici domande chiedi in chat su Telegram...non ha senso aprire un topic ad hoc per queste cose.

**Comment by Robertomastro** [comment]
mi manca il comando per verificare che le segnalazioni non ricompaiano.
Inoltre adesso ho il problema che dopo l'installazione di HA non vedo il tab del supervisor.
Cosa posso aver sbagliato?
grazie

**Reply by snakuzzo** [reply]
Mamma mia...è scritto su :D

Se dai il comando docker-comlose e non hai warning...è tutto ok

Quanto a supervisor...se hai letto il changelog...è stato spostato sotto configurazione

**Comment by Robertomastro** [comment]
posso rifare lo step di installazione senza problemi?

inoltre, ho provato a chiedere in chat:

ciao, sto provando a reinstallare da zero HA, seguendo la guida sono arrivato al passaggio sudo apt-get upgrade -y
ma ottengo il seguente errore:
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 vlc-bin : Depends: libvlc-bin (= 3.0.16-1+rpi1+rpt1) but 3.0.16-1+rpi1+rpt2 is to be installed
 vlc-plugin-skins2 : Depends: vlc-plugin-qt (= 3.0.16-1+rpi1+rpt2) but 3.0.16-1+rpi1+rpt1 is to be installed
E: Broken packages

posso proseguire o devo fare qualcos'altro per risolvere?

però non ho avuto risposta.
riprovo qui.
grazie

**Reply by snakuzzo** [reply]
Scusa ma da dove sono usciti fuori quei pacchetti ?

**Comment by Robertomastro** [comment]
sono usciti quando ho lanciato il comando
sudo apt-get upgrade -y
... coe scritto sopra :-)

**Reply by snakuzzo** [reply]
quella roba non c'entra niente con HA

**Comment by Robertomastro** [comment]
l'ho scritto, stavo ripatendo da zero.

**Comment by Robertomastro** [comment]
ciao di nuovo.
ho completato la nuova installazione. Continuo a non vedere "supervisor"
credo che dipenda che ho la versione 
Home Assistant 2021.11.5
ho provato ad aggiornare tramite ssh:

pi@raspberrypi:~/esempio-compose $ **docker-compose pull home-assistant**
Pulling home-assistant ... done

pi@raspberrypi:~/esempio-compose $ **docker-compose up --build -d home-assistant**
**_home-assistant is up-to-date_**

pi@raspberrypi:~/esempio-compose $ d**ocker image prune -f**
Total reclaimed space: 0B

pi@raspberrypi:~/esempio-compose $

ma non si aggiorna.
qualche suggerimento?


PS: posso rinominare la dir "esempio-compose"?
grazie

**Reply by snakuzzo** [reply]
Ma con questa installazione non hai il supervisor!!!
L'installazione che hai fatto è Core su Docker (detta anche Home Assistant Container)

LEGGI -> https://indomus.it/focus/home-assistant-hassio-o-home-assistant-core-quale-installazione-fa-al-caso-mio/

**Reply by Robertomastro** [reply]
avevo letto la guida.
mi sembrava che, a costo di qualche fatica in più, avrei avuto la versione migliore:

_CONCLUDENDO
In poche parole? Se potete, usate CORE – prendete le distanze da chi vi consiglia diversamente._

Quindi non posso avere Supervisor all'interno di Home Assistant?

la versione 2021.12 non è ancora disponibile per l'installzione Core su Doker?

grazie


PS: posso rinominare la dir "esempio-compose"?

**Reply by snakuzzo** [reply]
Infatti è la soluzione migliore...a nostro avviso
Avere il supervisor non vuol dire avere la versione migliore :D

Sì che è disponibile la versione per core, ma per le immagini di  RPI3 ed RPI4 hanno fatto casino con i tag delle immagini...quindi al posto del tag `stable` se vuoi (per ora) puoi passargli esplicitamente `2021.12.1`

P.S. La dir in cui hai il file `docker-compose.yaml` la puoi rinominare come ti pare (a patto che tu non l'abbia esplicitamente messa in qualche tuo script o cose simili)

**Reply by Robertomastro** [reply]
1) ho fatto:

- docker-compose pull home-assistant
- docker-compose up --build -d home-assistant 
- docker image prune -f
- sudo docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v /home/homeassistant/.homeassistant:/config --net=host --restart=always ghcr.io/home-assistant/home-assistant:2021.12.1

anche se il comando run mi sembra sia riferito alla versione “DOCKER RUN”.

mi restituisce:
**Unable to find image 'ghcr.io/home-assistant/home-assistant:2021.12.1' locally
2021.12.1: Pulling from home-assistant/home-assistant
Digest: sha256:ea818ffe2f8348bf2be290981628239042be3d53d5c26713603aac6847829730
Status: Downloaded newer image for ghcr.io/home-assistant/home-assistant:2021.12.1
docker: Error response from daemon: Conflict. The container name "/home-assistant" is already in use by container "eeeee13637b947d8c14f3299b3026762581764047580291ae13481b3fe5bc2b8". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.**

2) pensavo di rinominare la dir in docker ma ancora non l'ho fatto.

**Reply by snakuzzo** [reply]
Stai facendo un casino esagerato...
O usi `docker` o `docker-compose` ... deciditi. Stai mischiando alici e nutella... 

Usa docker compose e lascia perdere docker run

L'errore ce l'hai perché hai già creato un container con quel nome e la dir non c'entra un tubo!

Cancella il container che hai creato (fermalo se è in esecuzione

```
docker container stop home-assistant
docker container rm home-assistant`
```

poi con la situazione pulita userai solo `docker-compose`

P.S. `docker image prune -f` rimuove le immagini scaricate e non utilizzate per alcun container.

**Reply by snakuzzo** [reply]
E LEGGI I LOG... perché è tutto scritto lì :)

**Comment by Robertomastro** [comment]
ma sei tu che mi hai scritto di cambiare il tag:

**_Sì che è disponibile la versione per core, ma per le immagini di RPI3 ed RPI4 hanno fatto casino con i tag delle immagini...quindi al posto del tag stable se vuoi (per ora) puoi passargli esplicitamente 2021.12.1_**

e questo lo posso cambiare solo sul comando run.
... almeno dalle guide che ho letto.

MI dici il comando sul quale devo forzare il caricamento della 2021.12.1?

**Reply by snakuzzo** [reply]
@Robertomastro non è per cattiveria, ma dovresti andarti a riguardare come funziona docker e come funziona docker-compose...non puoi andare avanti a tentativi senza sapere quello che stai facendo.

Il tag lo imposti anche nel docker-compose.yaml in corrispondenza del service di homeassistant. Al posto di `stable` utilizzi`2021.12.1` (almeno finché non sistemano).

**Reply by Robertomastro** [reply]
adesso è chiaro.
Grazie
almeno la versione l'ho messa a posto
